packages/vm/src/crypto.rs

```diff
@@ -0,0 +1,177 @@
+use k256::{
+    ecdsa::signature::{DigestVerifier, Signature as _}, // traits
+    ecdsa::{Signature, VerifyingKey},                   // type aliases
+};
+use sha2::Digest; // trait
+
+use crate::dummy_sha2;
+use crate::errors::{VmError, VmResult};
+
+pub fn secp256k1_verify(message_hash: &[u8], signature: &[u8], public_key: &[u8]) -> VmResult<()> {
```

Comment:
This should probably return `VmResult<bool>`, where all signature verifications end up as `Ok(is_valid)` and only unexpected errors are `Err`. Then the ok value can be returned to the contract and errors terminate contract execution.

Reply:
I thought about that but decided against it, just to follow the underlying impl, which returns `Err` both when there's an error, and for failed verifications.

I can change it, though it may end up being a little kludgy, as I would basically have to inspect the error messages.